### PR TITLE
fix: namespace_selector invalid when restarting

### PR DESCRIPTION
**Pull request by AlinsRan**
### Type of change:

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

* When all namespace do not have  labels(namespace_selector), all resources will be monitored, and the namespace_selector function will not be available.
image

e2e-test:
* **[un-label]** When restarting, all ns without labels will be monitored
* **[ingress-controller restart]** When restarting, all ns without labels will be monitored

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: test/e2e/scaffold/ingress.go [hunk 1]
```diff
@@ -428,13 +428,17 @@ func (s *Scaffold) newIngressAPISIXController() error {
 	})
 
 	var ingressAPISIXDeployment string
-	label := fmt.Sprintf("apisix.ingress.watch=%s", s.namespace)
+	label := "\"\""
```
Comment (lingsamuel) [inline test/e2e/scaffold/ingress.go:431]: `` `""` `` would be clearer

**Review by lingsamuel — APPROVED** [review]

File: test/e2e/scaffold/ingress.go (hunk 1 above)
Comment (tao12345666333) [reply test/e2e/scaffold/ingress.go:431]: +1

File: test/e2e/suite-ingress/suite-ingress-features/namespace.go [hunk 2]
```diff
@@ -37,12 +38,37 @@ type headers struct {
...
 
 	ginkgo.Context("with namespace_selector", func() {
+		namespace1 := fmt.Sprintf("namespace-selector-1-%d", time.Now().Nanosecond())
+		namespace2 := fmt.Sprintf("namespace-selector-2-%d", time.Now().Nanosecond())
+
+		createNamespaceLabel := func(namespace string) {
+			k8s.CreateNamespaceWithMetadata(ginkgo.GinkgoT(), &k8s.KubectlOptions{ConfigPath: scaffold.GetKubeconfig()}, metav1.ObjectMeta{Name: namespace, Labels: s.NamespaceSelectorLabel()})
+			_, err := s.NewHTTPBINWithNamespace(namespace)
+			time.Sleep(6 * time.Second)
+			assert.Nil(ginkgo.GinkgoT(), err, "create second httpbin service")
```
Comment (tao12345666333) [inline test/e2e/suite-ingress/suite-ingress-features/namespace.go:59]:
```suggestion
			assert.Nil(ginkgo.GinkgoT(), err, "create new httpbin service")
```

File: test/e2e/suite-ingress/suite-ingress-features/namespace.go [hunk 3]
```diff
@@ -189,3 +269,130 @@ spec:
 		})
 	})
 })
+
+var _ = ginkgo.Describe("suite-ingress-features: namespacing ub-label", func() {
```
Comment (tao12345666333) [inline test/e2e/suite-ingress/suite-ingress-features/namespace.go:273]:
```suggestion
var _ = ginkgo.Describe("suite-ingress-features: namespacing un-label", func() {
```

File: test/e2e/suite-ingress/suite-ingress-features/namespace.go [hunk 4]
```diff
@@ -189,3 +269,130 @@ spec:
...
+kind: Namespace
+metadata:
+  name: %s
+`, namespace1)
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ns), "creating namespace")
+		assert.Nil(ginkgo.GinkgoT(), err, "create httpbin service in", namespace1)
+		time.Sleep(6 * time.Second)
+		routes, err := s.ListApisixRoutes()
+		assert.Nil(ginkgo.GinkgoT(), err)
+		assert.Len(ginkgo.GinkgoT(), routes, 0)
```
Comment (tao12345666333) [inline test/e2e/suite-ingress/suite-ingress-features/namespace.go:337]: In order to be more in line with the actual situation, I suggest modifying the logic here to directly clean up the namespace labels

Comment (AlinsRan) [reply test/e2e/suite-ingress/suite-ingress-features/namespace.go:337]: There is no such interface. This can be done, but it is not convenient.

Comment (tao12345666333) [reply test/e2e/suite-ingress/suite-ingress-features/namespace.go:337]: https://pkg.go.dev/k8s.io/client-go@v0.25.0/kubernetes/typed/core/v1#NamespaceInterface

This function can be done through Update or Patch

Comment (tao12345666333) [reply test/e2e/suite-ingress/suite-ingress-features/namespace.go:337]: Patch is simpler, just need to pass one of the following

```
 {"metadata":{"labels":{"YOUR Label Name":null}}}
```

Comment (AlinsRan) [reply test/e2e/suite-ingress/suite-ingress-features/namespace.go]: Done.

File: test/e2e/suite-ingress/suite-ingress-features/namespace.go [hunk 5]
```diff
@@ -90,18 +143,48 @@ spec:
...
 `, backendSvc, backendSvcPort[0])
+			assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(route3), "creating ingress")
+			time.Sleep(6 * time.Second)
+			routes, err = s.ListApisixRoutes()
+			assert.Nil(ginkgo.GinkgoT(), err)
+			assert.Len(ginkgo.GinkgoT(), routes, 1)
+			_ = s.NewAPISIXClient().GET("/headers").WithHeader("Host", "local.httpbin.org").Expect().Status(http.StatusNotFound)
+
+			// remove route1
+			assert.Nil(ginkgo.GinkgoT(), s.DeleteResourceFromStringWithNamespace(route1, namespace1), "delte ingress")
```
Comment (tao12345666333) [inline test/e2e/suite-ingress/suite-ingress-features/namespace.go:154]:
```suggestion
			assert.Nil(ginkgo.GinkgoT(), s.DeleteResourceFromStringWithNamespace(route1, namespace1), "delete ingress")
```

File: test/e2e/suite-ingress/suite-ingress-features/namespace.go [hunk 6]
```diff
@@ -90,18 +143,48 @@ spec:
...
+			_ = s.NewAPISIXClient().GET("/ip").WithHeader("Host", "httpbin.com").Expect().Status(http.StatusOK)
+			_ = s.NewAPISIXClient().GET("/headers").WithHeader("Host", "httpbin.org").Expect().Status(http.StatusNotFound)
+			_ = s.NewAPISIXClient().GET("/headers").WithHeader("Host", "local.httpbin.org").Expect().Status(http.StatusNotFound)
 		})
 	})
 })
 
-var _ = ginkgo.Describe("suite-ingress-features: namespacing filtering disable", func() {
-	s := scaffold.NewDefaultScaffold()
+var _ = ginkgo.Describe("suite-ingress-features: namespacing fltering disable", func() {
```
Comment (tao12345666333) [inline test/e2e/suite-ingress/suite-ingress-features/namespace.go:180]:
```suggestion
var _ = ginkgo.Describe("suite-ingress-features: namespacing filtering disable", func() {
```

File: test/e2e/suite-ingress/suite-ingress-features/namespace.go [hunk 7]
```diff
@@ -189,3 +271,134 @@ spec:
...
+		// un-label
+		_, err = client.CoreV1().Namespaces().Update(
+			context.Background(),
+			&v1.Namespace{ObjectMeta: metav1.ObjectMeta{
+				Name:   namespace1,
+				Labels: map[string]string{},
+			}},
+			metav1.UpdateOptions{},
+		)
+		assert.Nil(ginkgo.GinkgoT(), err, "creating namespace")
```
Comment (tao12345666333) [inline test/e2e/suite-ingress/suite-ingress-features/namespace.go:339]:
```suggestion
		assert.Nil(ginkgo.GinkgoT(), err, "unlabel the namespace")
```

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**